third-culture-software/bhima
# Fix XSL export problems

Fixed Excel export/download for
- Articles in Stock (Note: changed 'xlsxReport' to 'xlsx' until [Issue 6313](https://github.com/IMA-WorldHealth/bhima/issues/6313) is fixed)
- Stock Movements
- Stock Lots 

Added more hidden columns to these reports and purged some unneeded fields for these reports. This PR also fixes the errors when some of the fields were converted to Excel output (some date values were invalid; converted the invalid date to an empty string).

This is a reapplication of PR https://github.com/IMA-WorldHealth/bhima/pull/6304 and https://github.com/IMA-WorldHealth/bhima/pull/6314 to master (they were applied to sprint-prosani).

Closes https://github.com/IMA-WorldHealth/bhima/issues/6295
Closes https://github.com/IMA-WorldHealth/bhima/issues/6296

Note: This PR does not address XLS export from the Inventory Registry because it does not currently have an XLS export function.

**TESTING**
- Use any data set (but may be simpler with bhima_test)
- Try Stock Lots Registry, Articles in Stock, and Stock movements.  For each:
  - Try Menu > Columns and show all hidden columns
     - Verify that the columns make sense
  - Try Menu > Download as Excel
     - Open the downloaded file
       - Verify all column headers are translated (Capitalized)
       - Check data in all columns
- Try doing the same tests with a production data set
